### PR TITLE
feat: auth/credential support

### PR DESCRIPTION
**Pull request by venjiang**

### CODE REVIEW

File: example/go.mod [hunk 1]
```diff
@@ -7,3 +7,5 @@ require (
 	github.com/yomorun/yomo v1.7.2
 	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
 )
+
+replace github.com/yomorun/yomo => ../../yomo
```
Comment (fanweixiao) [inline example/go.mod:11]: should remove this line

File: example/source/main.go [hunk 2]
```diff
@@ -17,7 +17,11 @@ type noiseData struct {
 
 func main() {
 	// connect to YoMo-Zipper.
-	source := yomo.NewSource("yomo-source", yomo.WithZipperAddr("localhost:9000"))
+	source := yomo.NewSource(
+		"yomo-source",
+		yomo.WithZipperAddr("localhost:9000"),
+		yomo.WithCredential("token:1234"),
```
Comment (fanweixiao) [inline example/source/main.go:23]: Suggest read `YOMO_TOKEN` in `NewSource` automatically.

File: go.mod [hunk 3]
```diff
@@ -15,3 +15,5 @@ require (
 	golang.org/x/tools v0.1.10
 	gopkg.in/ini.v1 v1.66.4 // indirect
 )
+
+replace github.com/yomorun/yomo => ../yomo
```
Comment (fanweixiao) [inline go.mod:19]: should remove this line.

File: serverless/golang/serverless.go [hunk 4]
```diff
@@ -50,10 +51,15 @@ func (s *GolangServerless) Init(opts *serverless.Options) error {
 	}
 
 	// append main function
+	credential := viper.GetString("credential")
+	if len(credential) > 0 {
+		log.InfoStatusEvent(os.Stdout, "Credential=%s", credential)
```
Comment (fanweixiao) [inline serverless/golang/serverless.go:56]: should remove this log for security reason.

File: serverless/js/serverless.go [hunk 5]
```diff
@@ -44,10 +45,15 @@ func (s *JsServerless) Init(opts *serverless.Options) error {
 	}
 
 	// append main function
+	credential := viper.GetString("credential")
+	if len(credential) > 0 {
+		log.InfoStatusEvent(os.Stdout, "Credential=%s", credential)
```
Comment (fanweixiao) [inline serverless/js/serverless.go:50]: should remove this log for security reason.

File: cmd/serve.go [hunk 6]
```diff
@@ -44,6 +46,20 @@ var serveCmd = &cobra.Command{
 		if err != nil {
 			log.FailureStatusEvent(os.Stdout, err.Error())
 		}
+		// auth
+		auth := viper.GetString("auth")
+		if len(auth) > 0 {
+			idx := strings.Index(auth, ":")
```
Comment (fanweixiao) [inline cmd/serve.go:52]: Will a `AuthorizationFunc` is better?

**Review by fanweixiao — CHANGES_REQUESTED** [review]

File: example/go.mod (hunk 1 above)
Comment (venjiang) [reply example/go.mod:11]: Yeah, I known，It need review after `YoMo` PR merged.

File: serverless/golang/serverless.go (hunk 4 above)
Comment (venjiang) [reply serverless/golang/serverless.go:56]: Get

File: example/source/main.go (hunk 2 above)
Comment (venjiang) [reply example/source/main.go:23]: “token:1234” is credential payload,It’s can be anything. eg: “jwt:ugghjytffhj”

**Review by wujunzhuo — APPROVED** [review]

**Review by xiaojian-hong — APPROVED** [review]

**Review by fanweixiao — APPROVED** [review]

**Pull request merged**